libraries/apollo-runtime/src/appleMain/kotlin/com/apollographql/apollo3/network/http/DefaultHttpEngine.apple.kt

```diff
@@ -48,33 +48,16 @@ import platform.posix.pthread_mutex_lock
 import platform.posix.pthread_mutex_t
 import platform.posix.pthread_mutex_unlock
 
-/**
- * An [HttpEngine] based on [NSURLSession] with the ability to stream data as it is received when using `Transfer-Encoding: Chunked`.
- * This is useful when using `@defer`.
- *
- * Note: this implementation only works when using the [new native memory manager](https://github.com/JetBrains/kotlin/blob/master/kotlin-native/NEW_MM.md)
- * as multiple threads mutate the internal state.
- *
- * That is why it is considered experimental and subject to change.
- *
- * Using the default [DefaultHttpEngine] will work when using `@defer`, but the payloads are buffered until the response is fully
- * received, thus losing the incremental nature of the response.
- */
-@ApolloExperimental
-class StreamingNSURLSessionHttpEngine(
+actual class DefaultHttpEngine(
```

Comment:
Can you remove the expect/actual classes to use top level constructor functions instead? (Can be made in a separate PR)

Reply:
Will do it in the next PR (also to implement `Closeable` to replace `dispose()`)